apps/cms/src/lib/cms/client/metadata-execution.ts

```diff
@@ -39,11 +39,94 @@ export const executeMetadataUpdate = async ({
   if (Object.keys(cmsProviderInstanceIdsToDelete).length) {
     await apiClient
       .mutation(DeleteMetadataDocument, {
-        id: productVariant.id,
+        id: itemId,
         keys: Object.entries(cmsProviderInstanceIdsToDelete).map(([cmsProviderInstanceId]) =>
           createCmsKeyForSaleorItem(cmsProviderInstanceId)
         ),
       })
       .toPromise();
   }
 };
+
+export const executeMetadataUpdate = async ({
+  context,
+  productVariant,
+  cmsProviderInstanceIdsToCreate,
+  cmsProviderInstanceIdsToDelete,
+}: {
+  context: WebhookContext;
+  productVariant: WebhookProductVariantFragment;
+  cmsProviderInstanceIdsToCreate: Record<string, string>;
+  cmsProviderInstanceIdsToDelete: Record<string, string>;
+}) => {
+  const { token, saleorApiUrl } = context.authData;
+  const apiClient = createClient(saleorApiUrl, async () => ({ token }));
+
+  await executeMetadataUpdateMutation({
+    apiClient,
+    itemId: productVariant.id,
+    cmsProviderInstanceIdsToCreate,
+    cmsProviderInstanceIdsToDelete,
+  });
+};
+
+type ItemMetadataRecord = {
+  id: string;
+  cmsProviderInstanceIds: MetadataRecord;
+};
+
+export const executeBatchMetadataUpdate = async ({
```

Comment:
Nothing crucial, but I wonder about the `execute` naming convention. Why not just `batchUpdateMetadata` or something similar?